src/main/kotlin/platformapi/channelDetails.kt

```diff
@@ -0,0 +1,43 @@
+package platformapi
+
+import io.ktor.application.*
+import io.ktor.http.*
+import io.ktor.request.*
+import io.ktor.response.*
+import io.ktor.util.pipeline.*
+import persistence.jooq.KotlinDslContext
+import persistence.postgres.queries.*
+import platformapi.models.ChannelWrite
+import util.Fallible
+
+suspend fun PipelineContext<Unit, ApplicationCall>.upsertChannel(
+    location: ChannelList.ChannelDetails,
+    database: KotlinDslContext
+) {
+    val channel = call.receive<ChannelWrite>()
+    // Have two transactions as postgres does not like reusing a transaction that
+    // had errors. (https://stackoverflow.com/questions/10399727/psqlexception-current-transaction-is-aborted-commands-ignored-until-end-of-tra)
+    when (database.transaction {
+        val insertResult = insertChannel(channel.meta)
+        insertMembers(channel.members)
+        insertResult
+    }) {
+        is Fallible.Failure -> {
+            database.transaction {
+                updateChannel(channel.meta)
+                deleteMembersOf(location.channelId)
+                insertMembers(channel.members)
```

Comment:
I am just scrolling through this out of curiosity but shouldn't you check for `InsertChannelError.Duplicate` in that case? 🤔

Reply:
Nice catch, didn't notice that.